exonum/src/events/noise/tests.rs

```diff
@@ -410,25 +423,27 @@ impl NoiseErrorHandshake {
 }
 
 impl Handshake for NoiseErrorHandshake {
-    fn listen<S>(self, stream: S) -> HandshakeResult<S>
+    type Result = Option<ConnectInfo>;
```

Comment:
I'd say that `Option`al nature of the result could be better encapsulated into the error variants of `listen` / `send` results. Or is it a normal that the information cannot be obtained from the remote peer?

Reply:
It's normal for the sender to not receive information from responder during the handshake because we have two connections between peers. The sender receives it when responder creates a return connection.